tripy/tests/test_examples.py

```diff
@@ -80,12 +80,43 @@ def __str__(self):
         return os.path.relpath(self.path, EXAMPLES_ROOT)
 
 
-EXAMPLES = [Example(["nanogpt"])]
+EXAMPLES = [Example(["nanogpt"]), Example(["segment-anything-model-v2"])]
 
 
 @pytest.mark.l1
 @pytest.mark.parametrize("example", EXAMPLES, ids=lambda case: str(case))
 def test_examples(example, sandboxed_install_run):
+
+    def test_with_tolerance(number, value, tolerance):
+        tolerance = float(tolerance) / 100
+        lower = float(number) * (1 - tolerance)
+        upper = float(number) * (1 + tolerance)
+        try:
+            return lower <= float(value) <= upper
+        except ValueError:
+            return False
+
+    def process_tolerances(expected_output):
+        specs = []
+        placeholder_regex = r"{(\d+\.?\d*)±(\d+)%}"
+        pattern = expected_output
+
+        # Replace tolerance patterns with more flexible capture group
+        matches = list(re.finditer(placeholder_regex, pattern))
+        for match in matches:
+            specs.append((match.group(1), match.group(2)))
+            pattern = pattern.replace(match.group(0), r"(\d+\.?\d*)", 1)
+
+        # Escape parentheses but not our capture group
+        pattern = pattern.replace("(", r"\(")
+        pattern = pattern.replace(")", r"\)")
+        pattern = pattern.replace(r"\(\d+\.?\d*\)", r"(\d+\.?\d*)")
+
+        # Make whitespace flexible
+        pattern = pattern.replace(" ", r"\s+")
```

Comment:
I don't quite understand what's happening here. Can we encode the flexibility into the regex itself?